jsonrpc/dispatcher.go

```diff
@@ -242,20 +243,24 @@ func (d *Dispatcher) Handle(reqBody []byte) ([]byte, error) {
 		if err := json.Unmarshal(reqBody, &req); err != nil {
 			return nil, invalidJSONRequest
 		}
+		if req.Method == "" {
+			//TODO:Add id if required
+			return nil, invalidJSONRequest
+		}
 		return d.handleReq(req)
 	}
 
 	// handle batch requests
 	var requests []Request
 	if err := json.Unmarshal(reqBody, &requests); err != nil {
-		return nil, d.internalError("batch method", err)
+		return nil, d.internalError(nil, "batch method", err)
 	}
-	var responses []Response
+	var responses []interface{}
```

Comment:
I would really try to step away from using this interface umbrella for multiple different response types. It would be great if you could create a common interface that ErrorResponse and Response could implement.